packages/loader/container-loader/src/loader.ts

```diff
@@ -412,3 +392,86 @@ export class Loader implements IHostLoader {
 		);
 	}
 }
+
+/**
+ * Loads container and leaves it in a state where it does not process any ops
+ * If sequence number is provided, loads up to this sequence number and stops there, otherwise stops immediately after loading snapshot.
+ * @internal
+ * */
+export async function loadContainerPaused(
+	loader: ILoader,
+	request: IRequest,
+	loadToSequenceNumber?: number,
+) {
+	const container = await loader.resolve({
+		url: request.url,
+		headers: {
+			...Headers,
+			// ensure we do not process any ops.
+			[LoaderHeader.loadMode]: { opsBeforeReturn: undefined, deltaConnection: "none" },
+		},
+	});
+
+	// If we are trying to pause at a specific sequence number, ensure the latest snapshot is not newer than the desired sequence number.
+	if (loadToSequenceNumber !== undefined) {
+		const lastProcessedSequenceNumber = container.deltaManager.initialSequenceNumber;
+		if (lastProcessedSequenceNumber > loadToSequenceNumber) {
+			throw new Error(
+				"Cannot satisfy request to pause the container at the specified sequence number. Most recent snapshot is newer than the specified sequence number.",
+			);
+		}
+	}
+
+	// Force readonly mode - this will ensure we don't receive an error for the lack of join op
+	container.forceReadonly?.(true);
+
+	// We need to setup a listener to stop op processing once we reach the desired sequence number (if specified).
+	const opHandler = () => {
+		if (loadToSequenceNumber === undefined) {
+			// If there is no specified sequence number, pause after the inbound queue is empty.
+			if (container.deltaManager.inbound.length !== 0) {
+				return;
+			}
+		} else {
+			// If there is a specified sequence number, keep processing until we reach it.
+			if (container.deltaManager.lastSequenceNumber < loadToSequenceNumber) {
+				return;
+			}
+		}
+
+		// Pause op processing once we have processed the desired number of ops.
+		void container.deltaManager.inbound.pause();
+		void container.deltaManager.outbound.pause();
+		container.off("op", opHandler);
+	};
+	if (
+		(loadToSequenceNumber === undefined && container.deltaManager.inbound.length === 0) ||
+		container.deltaManager.lastSequenceNumber === loadToSequenceNumber
+	) {
+		// If we have already reached the desired sequence number, call opHandler() to pause immediately.
+		opHandler();
+	} else {
+		// If we have not yet reached the desired sequence number, setup a listener to pause once we reach it.
+		container.on("op", opHandler);
+	}
+
+	container.connect();
```

Comment:
Should this always connect?  E.g. probably not if the original request specified an alternate deltaConnection behavior.

Reply:
If we do not do it, there is a chance that we will not get any ops for quite a while. Protocol does not describe frequency at which ordering service flushes ops to storage - it's responsibility of a client to ask both storage and be connected to delta connection to guarantee forward progress. Driver does not have to implement ops caching (same for browser - it is free to cleanup localStorage, or even reject app from to use it - happens in some privacy mode).
PUSH flushes ops to storage only on summaries.

Old flow did it too, it was just not obvious (call to Container.handleDeltaConnectionArg()), but it also had an issue that user could have provided IContainerLoadMode options that would prevent any ops fetching (thus deadlocking process). This function overwrites IContainerLoadMode.

Reply:
Also added a comment above it on why it's required.

Reply:
This should at least have a tsdoc comment for `loadContainerPaused` saying that "LoaderHeader.loadMode is not supported when using this function".  And probably a UsageError to ensure the customer won't be surprised when their option is overridden.

Another option would be to detect that we have run out of ops to process without connecting and throw, or to just let it sit pending forever.  Both of these options are more-obviously explainable to a customer than connecting when they explicitly requested not to.